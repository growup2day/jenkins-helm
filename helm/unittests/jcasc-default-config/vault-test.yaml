templates:
  - config-jcasc.yaml
tests:
  - it: should include vault config in default config scripts when configured
    set:
      JCasC:
        defaultConfig: true
        vault:
          credentials:
            artifactory-api-key:
              kind: vaultUsernamePasswordCredentialImpl
              scope: GLOBAL
              description: "API Key for Artifactory"
              path:
                'secret/context/{{ .Release.Namespace }}/{{ include "jenkins.serviceAccountName" .
                }}/pairs/artifactory-api-key'
            fortify-ci-token:
              kind: vaultStringCredentialImpl
              scope: GLOBAL
              description: "CI Token for Fortify"
              path: 'secret/context/{{ .Release.Namespace }}/{{ include "jenkins.serviceAccountName" . }}/keys'
              vaultKey: "fortify-ci-token"
    asserts:
      - isNotNull:
          path: data.[jcasc-default-config-vault.yaml]
      - matchSnapshot:
          path: data.[jcasc-default-config-vault.yaml]
  - it: should include blank vault config in default config scripts when not configured
    set:
      JCasC:
        defaultConfig: true
    asserts:
      - isNotNull:
          path: data.[jcasc-default-config-vault.yaml]
      - matchSnapshot:
          path: data.[jcasc-default-config-vault.yaml]
  - it: should not include vault config in default config scripts when there is custom config script for the section
    set:
      JCasC:
        defaultConfig: true
        configScripts:
          custom-override: |
            unclassified:
              hashicorpVault:
                configuration:
                  vaultUrl: "https://dummy-vault.com"
    asserts:
      - isNull:
          path: data.[jcasc-default-config-vault.yaml]
      - isNotNull:
          path: data.[custom-override.yaml]
  - it: should not include vault config in default config scripts when disabled
    set:
      JCasC:
        defaultConfig: true
        vault:
          enabled: false
    asserts:
      - isNull:
          path: data.[jcasc-default-config-vault.yaml]
