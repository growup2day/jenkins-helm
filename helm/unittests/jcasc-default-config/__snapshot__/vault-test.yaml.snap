should include blank vault config in default config scripts when not configured:
  1: |
    |-
      unclassified:
        # Configure System > Vault Plugin
        hashicorpVault:
          configuration:
            vaultCredentialId: ""
            vaultNamespace: ""
            vaultUrl: ""
should include vault config in default config scripts when configured:
  1: |
    |-
      unclassified:
        # Configure System > Vault Plugin
        hashicorpVault:
          configuration:
            vaultCredentialId: ""
            vaultNamespace: ""
            vaultUrl: ""
